components/brave_ads/browser/BUILD.gn

```diff
@@ -39,25 +38,28 @@ static_library("browser") {
   deps = [
     "//base",
     "//brave/components/brave_adaptive_captcha",
-    "//brave/components/brave_ads/core",
     "//brave/components/brave_ads/resources",
     "//brave/components/brave_component_updater/browser",
     "//brave/components/l10n/common",
+    "//brave/components/ntp_background_images/common",
     "//brave/components/p3a_utils",
     "//brave/components/time_period_storage",
-    "//components/keyed_service/core",
     "//components/pref_registry",
     "//components/prefs",
     "//components/sessions",
-    "//sql",
     "//third_party/blink/public/common",
+    "//ui/base/idle",
     "//url",
   ]
 
-  public_deps = [ "//brave/components/brave_ads/core/mojom" ]
-
-  # TODO(https://github.com/brave/brave-browser/issues/24163)
-  check_includes = false
+  public_deps = [
+    "application_state",
+    "//brave/components/brave_ads/core",
+    "//brave/components/brave_ads/core/mojom",
+    "//brave/components/services/bat_ads/public/interfaces",
+    "//components/keyed_service/core",
+    "//mojo/public/cpp/bindings",
+  ]
 
```

Comment:
`"//chrome/browser/notifications"` dependency was removed from AdsServiceImpl, but it is still listed here below

Reply:
That's odd, I wonder how it's passing gn check since I removed all the `specific_include_rules`